crypto-ffi/src/wasm/mod.rs

```diff
@@ -79,96 +79,201 @@ pub extern "C" fn __stack_chk_fail() {
+    BufferedFutureMessage,
+    #[error("Incoming message is from an epoch too far in the future to buffer.")]
+    WrongEpoch,
+    #[error("The epoch in which message was encrypted is older than allowed")]
+    MessageEpochTooOld,
+    #[error("Tried to decrypt a commit created by self which is likely to have been replayed by the DS")]
+    SelfCommitIgnored,
+    #[error(
+        "You tried to join with an external commit but did not merge it yet. We will reapply this message for you when you merge your external commit"
+    )]
+    UnmergedPendingGroup,
+    #[error("The received proposal is deemed stale and is from an older epoch.")]
+    StaleProposal,
+    #[error("The received commit is deemed stale and is from an older epoch.")]
+    StaleCommit,
+    #[error("Another MLS error occurred but the details are probably irrelevant to clients")]
+    Other,
+}
+
+impl From<core_crypto::MlsError> for MlsError {
+    #[inline]
+    fn from(_: core_crypto::MlsError) -> Self {
+        Self::Other
+    }
+}
+
+#[cfg(feature = "proteus")]
+#[derive(Debug, thiserror::Error, strum::AsRefStr)]
+pub enum ProteusError {
+    #[error("The requested session was not found")]
+    SessionNotFound,
+    #[error("We already decrypted this message once")]
+    DuplicateMessage,
+    #[error("The remote identity has changed")]
+    RemoteIdentityChanged,
+    #[error("Another Proteus error occurred but the details are probably irrelevant to clients")]
+    Other(Option<u16>),
+}
+
+#[cfg(feature = "proteus")]
+impl From<core_crypto::ProteusError> for ProteusError {
+    fn from(value: core_crypto::ProteusError) -> Self {
+        type SessionError = proteus_wasm::session::Error<core_crypto_keystore::CryptoKeystoreError>;
+        match value {
+            core_crypto::ProteusError::ProteusSessionError(SessionError::InternalError(
+                proteus_wasm::internal::types::InternalError::NoSessionForTag,
+            )) => Self::SessionNotFound,
+            core_crypto::ProteusError::ProteusSessionError(SessionError::DuplicateMessage) => Self::DuplicateMessage,
+            core_crypto::ProteusError::ProteusSessionError(SessionError::RemoteIdentityChanged) => {
+                Self::RemoteIdentityChanged
+            }
+            _ => Self::Other(value.error_code()),
+        }
+    }
+}
+
+#[derive(Debug, thiserror::Error, strum::AsRefStr)]
+pub(crate) enum InternalError {
     #[error(transparent)]
-    CryptoError(#[from] CryptoError),
+    MlsError(#[from] MlsError),
+    #[cfg(feature = "proteus")]
     #[error(transparent)]
-    E2eError(#[from] E2eIdentityError),
+    ProteusError(#[from] ProteusError),
+    #[error("End to end identity error")]
+    E2eiError,
     #[error(transparent)]
     SerializationError(#[from] serde_wasm_bindgen::Error),
-    #[error("Failed lifting an enum")]
-    EnumError,
+    #[error("Unknown cyphersuite identifier")]
+    UnknownCyphersuite,
```

Comment:
I'm also a fan of British English, but would suggest to stick to the `ciphersuite` spelling, since that's what the rest of the code uses, both in Wire and the external projects. :-)